assets/js/layouts/playground.js

```diff
@@ -11,34 +9,37 @@ export class Playground {
 				e.preventDefault();
 				this.switchTo(link);
 			}
-		})
+		});
 	}
 
-	switchTo(link) {
+	async switchTo(link) {
 		// fade out the old image
-		this.$el.querySelector(".playground-header-figure-wrapper").classList.add("loading");
-
-		setTimeout(async () => {
-			const response = await fetch(link);
-			const body     = await response.text();
-			const parser   = new DOMParser();
-			const doc      = parser.parseFromString(body, "text/html");
-
-			// hide the new image before it gets injected
-			doc.querySelector(".playground-header-figure-wrapper").classList.add("loading");
-
-			// replace the playground
-			this.$el.innerHTML = doc.querySelector(".playground").innerHTML;
-
-			// fade in the image on load
-			this.$el.querySelector(".playground-header-figure img").addEventListener("load", function () {
-				setTimeout(() => {
-					this.parentNode.classList.remove("loading");
-				}, 200);
+		this.$el
+			.querySelector(".playground-header-figure-wrapper")
+			.classList.add("loading");
+
+		const response = await fetch(link);
+		const body = await response.text();
+		const parser = new DOMParser();
+		const doc = parser.parseFromString(body, "text/html");
+
+		// hide the new image before it gets injected
+		doc
+			.querySelector(".playground-header-figure-wrapper")
+			.classList.add("loading");
+
+		// replace the playground
+		this.$el.innerHTML = doc.querySelector(".playground").innerHTML;
+
+		// fade in the image on load
+		this.$el
+			.querySelector(".playground-header-figure img")
+			.addEventListener("load", async function () {
+				await new Promise((resolve) => setTimeout(resolve, 100));
```

Comment:
TBH I don't fully understand what this delay is even trying to do, also in the old version. We had one delay that waits until the old image is faded out and one delay that waits for another 200 (now 100) ms before fading the new image in. But the actual fading-in happens *after* that second delay. So why do we need to wait here?

What I think we need is logic like "perform the fetch immediately and if less than 100 ms (= fading-out time of the old image) has passed by the time we are ready, wait until the fade-out was completed". So basically a `Promise.all()` of the fetch request and a 100 ms delay. Once both are finished, the new content can be injected and faded in.

Reply:
You are right, and the `Promise.all()` sounds like a good direction. But I think it's not just fetch and the animation delay, but the image being loaded additionally.

Reply:
Ok, now fully understood it. Give me a few minutes.